pyanaconda/modules/network/initialization.py

```diff
@@ -240,89 +240,97 @@ def _run(self, nm_client):
+
+            if not dumped_con and len(initramfs_cons) == 1:
+                # Try to clone the persistent connection for the device
+                # from the connection which should be a generic (not bound
+                # to iface) connection created by NM in initramfs
+                dumped_con = clone_connection_sync(nm_client, initramfs_cons[0], con_id=iface)
+
+            if dumped_con:
                 log.debug("%s: dumping connection %s to config file for %s",
-                          self.name, con.get_uuid(), iface)
-                commit_changes_with_autoconnection_blocked(con, nm_client)
+                          self.name, dumped_con.get_uuid(), iface)
+                self._dump_connection(nm_client, dumped_con, iface, bool(initramfs_cons))
             else:
                 log.debug("%s: none of the connections can be dumped as persistent",
                           self.name)
-                if n_cons > 1 and not device_is_port:
+                if len(available_cons) > 1 and not device_is_port:
                     log.warning("%s: unexpected number of connections, not dumping any",
                                 self.name)
                     continue
                 log.debug("%s: creating default connection for %s", self.name, iface)
-                network_data = copy.deepcopy(self._default_network_data)
-                if has_initramfs_con:
-                    network_data.onboot = True
-                add_connection_from_ksdata(
-                    nm_client,
-                    network_data,
-                    iface,
-                    activate=False,
-                    ifname_option_values=self._ifname_option_values
-                )
+                self._create_default_connection(nm_client, iface, bool(initramfs_cons))
 
             new_configs.append(iface)
 
         return new_configs
 
+    def _dump_connection(self, nm_client, dumped_con, iface, initramfs_con):
+        self._update_connection(nm_client, dumped_con, iface)
+        # Update some values of connection generated in initramfs so it
+        # can be used as persistent configuration.
+        if initramfs_con:
+            update_connection_values(
+                dumped_con,
+                [
+                    # Make sure ONBOOT is yes
+                    (NM.SETTING_CONNECTION_SETTING_NAME,
+                        NM.SETTING_CONNECTION_AUTOCONNECT,
+                        True),
+                    # Update cloned generic connection from initramfs
+                    (NM.SETTING_CONNECTION_SETTING_NAME,
+                        NM.SETTING_CONNECTION_MULTI_CONNECT,
+                        0),
+                    # Update cloned generic connection from initramfs
+                    (NM.SETTING_CONNECTION_SETTING_NAME,
+                        NM.SETTING_CONNECTION_WAIT_DEVICE_TIMEOUT,
+                        -1)
```

Comment:
The indentation seems to be weird here. Is it correct?

Reply:
Thank you for the catch, updated.